tests/failpoints/cases/test_cmd_epoch_checker.rs

```diff
@@ -0,0 +1,373 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_rocks::RocksSnapshot;
+use kvproto::raft_cmdpb::{RaftCmdRequest, RaftCmdResponse};
+use raft::eraftpb::MessageType;
+use raftstore::store::msg::*;
+use test_raftstore::*;
+use tikv_util::HandyRwLock;
+
+use std::sync::mpsc::{self, TryRecvError};
+use std::time::Duration;
+
+struct CbReceivers {
+    proposed: mpsc::Receiver<()>,
+    committed: mpsc::Receiver<()>,
+    applied: mpsc::Receiver<RaftCmdResponse>,
+}
+
+impl CbReceivers {
+    fn assert_not_ready(&self) {
+        sleep_ms(100);
+        assert_eq!(self.proposed.try_recv().unwrap_err(), TryRecvError::Empty);
+        assert_eq!(self.committed.try_recv().unwrap_err(), TryRecvError::Empty);
+        assert_eq!(self.applied.try_recv().unwrap_err(), TryRecvError::Empty);
+    }
+
+    fn assert_ok(&self) {
+        let resp = self
+            .applied
+            .recv_timeout(Duration::from_millis(100))
```

Comment:
Hmmm, maybe the timeout is a little bit shorter and may lead to an unstable test.
Maybe 1 second is better.